From 424de0923cb3d44d010e3e2f9bd7bc75d5c30234 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gonzalo=20Tornar=C3=ADa?= <tornaria@cmat.edu.uy>
Date: Sun, 26 Dec 2021 22:34:39 -0300
Subject: [PATCH] sage-site: exec only if it is executable

---
 src/bin/sage | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/bin/sage b/src/bin/sage
index 4c0fa17426..153e610a2e 100755
--- a/src/bin/sage
+++ b/src/bin/sage
@@ -129,7 +129,7 @@ usage() {
     echo "                           --optional - controls which optional tests are run"
     echo "                           --help - show all testing options"
     echo "  -v, --version       -- display Sage version information"
-    if [ -n "$SAGE_ROOT" ]; then
+    if [ -n "$SAGE_ROOT" -a -x "$SAGE_ROOT/build/bin/sage-site" ]; then
         exec "$SAGE_ROOT/build/bin/sage-site" "-h"
     fi
     exit 0
@@ -555,7 +555,7 @@ usage_advanced() {
     echo "                         files are named sage-omega.PID can be found in"
     echo "                         \$DOT_SAGE"
     echo "  --valgrind          -- this is an alias for --memcheck"
-    if [ -n "$SAGE_ROOT" ]; then
+    if [ -n "$SAGE_ROOT" -a -x "$SAGE_ROOT/build/bin/sage-site" ]; then
         exec "$SAGE_ROOT/build/bin/sage-site" "--advanced"
     fi
     echo
@@ -1131,7 +1131,7 @@ case "$1" in
     -*)
         # Delegate further option handling to the non-installed sage-site script.
         # (These options become unavailable if the directory $SAGE_ROOT is removed.)
-        if [ -n "$SAGE_ROOT" ]; then
+        if [ -n "$SAGE_ROOT" -a -x "$SAGE_ROOT/build/bin/sage-site" ]; then
             exec "$SAGE_ROOT/build/bin/sage-site" "$@"
             # fallthrough if there is no sage-site script
         fi
-- 
2.34.1

