diff --git a/src/bin/sage-runtests b/src/bin/sage-runtests
index 288c889bd1..b9a4d3038a 100755
--- a/src/bin/sage-runtests
+++ b/src/bin/sage-runtests
@@ -15,7 +15,7 @@ os.environ['SAGE_STARTUP_FILE'] = os.path.join(DOT_SAGE, 'init-doctests.sage')
 
 def _get_optional_defaults():
     """Return the default value for the --optional flag."""
-    optional = ['sage', 'dochtml', 'optional']
+    optional = ['sage', 'optional']
 
     # This should be good-enough to determine if we are in a sage source
     # repository working tree
