From ee51c4322416b2ef96590a5b42d3916342f48059 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Sat, 18 Dec 2021 14:23:57 -0800
Subject: src/sage/all.py: Suppress numpy deprecation warnings triggered by
 pythran

---
 src/sage/all.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/sage/all.py b/src/sage/all.py
index 27cbfe7..3b50ef0 100644
--- a/src/sage/all.py
+++ b/src/sage/all.py
@@ -91,6 +91,10 @@ warnings.filterwarnings('ignore', category=DeprecationWarning,
 warnings.filterwarnings('ignore', category=DeprecationWarning,
     module='(.*[.]_vendor[.])?packaging')
 
+# Ignore numpy warnings triggered by pythran
+warnings.filterwarnings('ignore', category=DeprecationWarning,
+                        module='pythran')
+
 ################ end setup warnings ###############################
 
 
-- 
cgit v1.0-1-gd88e

