diff --git a/build/pkgs/pyzmq/checksums.ini b/build/pkgs/pyzmq/checksums.ini
index 3824fd2ce9..20509db618 100644
--- a/build/pkgs/pyzmq/checksums.ini
+++ b/build/pkgs/pyzmq/checksums.ini
@@ -1,5 +1,5 @@
 tarball=pyzmq-VERSION.tar.gz
-sha1=50abf6482a6395f15dce0b2774021e12c9f4faad
-md5=f261acf884460e93d30f7113c8e9242e
-cksum=1661206741
+sha1=d8bb8cb345c0bfd9e67a95ac796bf4b10354dde6
+md5=ecf13c72fbea05ba5ddc771295409d48
+cksum=1680251717
 upstream_url=https://pypi.io/packages/source/p/pyzmq/pyzmq-VERSION.tar.gz
diff --git a/build/pkgs/pyzmq/package-version.txt b/build/pkgs/pyzmq/package-version.txt
index 0398faf11c..8326e27f9b 100644
--- a/build/pkgs/pyzmq/package-version.txt
+++ b/build/pkgs/pyzmq/package-version.txt
@@ -1 +1 @@
-22.2.1
+22.3.0
diff --git a/build/pkgs/pyzmq/spkg-install.in b/build/pkgs/pyzmq/spkg-install.in
index 0ce404ee5a..069d87767f 100644
--- a/build/pkgs/pyzmq/spkg-install.in
+++ b/build/pkgs/pyzmq/spkg-install.in
@@ -1,17 +1,11 @@
-# Since we use environment vars we have to generate setup.cfg
-
-echo "[build_ext]" > src/setup.cfg
-
-# (I tried putting quotes around $SAGE_LOCAL to allow for spaces in
-# the path---which is never used but is a good habit to support---but
-# that simply does not work. Sorry.)
-
-echo "library_dirs = $SAGE_LOCAL/lib/" >> src/setup.cfg
-echo "include_dirs = $SAGE_LOCAL/include/" >> src/setup.cfg
-
-echo "[configure]" >> src/setup.cfg
-
 cd src
 
+# Generate setup.cfg
+cat > setup.cfg <<EOF
+[global]
+zmq_prefix = $SAGE_LOCAL
+no_libzmq_extension = True
+EOF
+
 # Configure and install
 sdh_pip_install .
diff --git a/build/pkgs/setuptools/checksums.ini b/build/pkgs/setuptools/checksums.ini
index 6a280871e7..d9290d04ff 100644
--- a/build/pkgs/setuptools/checksums.ini
+++ b/build/pkgs/setuptools/checksums.ini
@@ -1,5 +1,5 @@
 tarball=setuptools-VERSION.tar.gz
-sha1=5d97de0e774b2269c85685a4aa8fd5956bcfd2bb
-md5=ff20ab7e0d51c5ad0a9438c50e598c06
-cksum=3564252314
+sha1=cd4e5d1f7c47f86354fc89d040d869cde244da3b
+md5=79b36f59270a6f2e45d6a9a0c6dc68c1
+cksum=430889097
 upstream_url=https://pypi.io/packages/source/s/setuptools/setuptools-VERSION.tar.gz
diff --git a/build/pkgs/setuptools/package-version.txt b/build/pkgs/setuptools/package-version.txt
index e6ea636051..aec1257e4c 100644
--- a/build/pkgs/setuptools/package-version.txt
+++ b/build/pkgs/setuptools/package-version.txt
@@ -1 +1 @@
-58.2.0
+59.2.0
diff --git a/build/pkgs/zeromq/checksums.ini b/build/pkgs/zeromq/checksums.ini
index ea8c36bce5..dc802612dd 100644
--- a/build/pkgs/zeromq/checksums.ini
+++ b/build/pkgs/zeromq/checksums.ini
@@ -1,4 +1,5 @@
 tarball=zeromq-VERSION.tar.gz
-sha1=1132f839d703486c4ee1cf22f056585dfbb329c2
-md5=a1c95b34384257e986842f4d006957b8
-cksum=136261411
+sha1=47277a64749049123d1401600e8cfbab10a3ae28
+md5=c897d4005a3f0b8276b00b7921412379
+cksum=1500782345
+upstream_url=https://github.com/zeromq/libzmq/releases/download/vVERSION/zeromq-VERSION.tar.gz
diff --git a/build/pkgs/zeromq/package-version.txt b/build/pkgs/zeromq/package-version.txt
index df0228dfae..eda862a98c 100644
--- a/build/pkgs/zeromq/package-version.txt
+++ b/build/pkgs/zeromq/package-version.txt
@@ -1 +1 @@
-4.2.5
+4.3.4
diff --git a/build/pkgs/zeromq/patches/getrandom.patch b/build/pkgs/zeromq/patches/getrandom.patch
deleted file mode 100644
index e665114e1c..0000000000
--- a/build/pkgs/zeromq/patches/getrandom.patch
+++ /dev/null
@@ -1,39 +0,0 @@
-Patch taken from upstream git repo
-and after running aclocal && autoconf
-and removing changes to autotools input files
-
-commit 4ff814f204ee38177a392526cf12c8c0019b480f
-Author: Luca Boccassi <luca.boccassi@gmail.com>
-Date:   Wed May 9 10:41:20 2018 +0100
-
-    Problem: getrandom test does not check if it's working
-
-    Solution: check return value in autoconf and CMake. On some platforms
-    the function is available but not implemented (eg: GNU/Hurd).
-
-diff -ru zeromq-4.2.5//builds/cmake/Modules/ZMQSourceRunChecks.cmake b/builds/cmake/Modules/ZMQSourceRunChecks.cmake
---- zeromq-4.2.5//builds/cmake/Modules/ZMQSourceRunChecks.cmake	2018-03-23 20:33:36.000000000 +0100
-+++ b/builds/cmake/Modules/ZMQSourceRunChecks.cmake	2018-07-24 12:39:23.340035053 +0200
-@@ -287,7 +287,8 @@
- int main (int argc, char *argv [])
- {
-     char buf[4];
--    getrandom(buf, 4, 0);
-+    int rc = getrandom(buf, 4, 0);
-+    return rc == -1 ? 1 : 0;
- }
- "
-     ZMQ_HAVE_GETRANDOM)
-diff -ru zeromq-4.2.5//configure b/configure
---- zeromq-4.2.5//configure	2018-03-23 20:34:18.000000000 +0100
-+++ b/configure	2018-07-24 12:39:49.018035120 +0200
-@@ -23315,7 +23315,8 @@
- int main (int argc, char *argv [])
- {
-     char buf[4];
--    getrandom(buf, 4, 0);
-+    int rc = getrandom(buf, 4, 0);
-+    return rc == -1 ? 1 : 0;
- }
- 
- _ACEOF
