diff --git a/src/bin/sage-env b/src/bin/sage-env
index 27edbf4cfb..0ead5789cb 100644
--- a/src/bin/sage-env
+++ b/src/bin/sage-env
@@ -411,14 +411,6 @@ if [ -n "$PYTHONHOME" ]; then
     unset PYTHONHOME
 fi
 
-if [ -n "$SAGE_LOCAL" ]; then
-    LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
-    if [ "$UNAME" = "Linux" ]; then
-        LDFLAGS="-Wl,-rpath-link,$SAGE_LOCAL/lib $LDFLAGS"
-    fi
-    export LDFLAGS
-fi
-
 if [ -z "$IPYTHONDIR" ]; then
     # We hardcode a version number in the directory name. The idea is
     # that we keep using the same version number as long as that is
diff --git a/src/sage_setup/setenv.py b/src/sage_setup/setenv.py
index b059a6e709..b6a9e3646c 100644
--- a/src/sage_setup/setenv.py
+++ b/src/sage_setup/setenv.py
@@ -30,11 +30,6 @@ def setenv():
         _environ_prepend('PATH',         f'{SAGE_LOCAL}/bin')
         _environ_prepend('LIBRARY_PATH', f'{SAGE_LOCAL}/lib')
         _environ_prepend('CPATH',        f'{SAGE_LOCAL}/include')
-        _environ_prepend('LDFLAGS',      f'-L{SAGE_LOCAL}/lib -Wl,-rpath,{SAGE_LOCAL}/lib',
-                         separator=' ')
-        if UNAME == 'Linux':
-            _environ_prepend('LDFLAGS',      f'-Wl,-rpath-link,{SAGE_LOCAL}/lib',
-                             separator=' ')
         if Path(SAGE_VENV).resolve() != Path(SAGE_LOCAL).resolve():
             # This condition is always true, but we are keeping it for clarity.
             _environ_prepend('PATH',         f'{SAGE_VENV}/bin')
