More site packages not included in trac 29665

diff --git a/build/pkgs/gmpy2/spkg-configure.m4 b/build/pkgs/gmpy2/spkg-configure.m4
new file mode 100644
index 0000000000..2d0390f7db
--- /dev/null
+++ b/build/pkgs/gmpy2/spkg-configure.m4
@@ -0,0 +1 @@
+SAGE_SPKG_CONFIGURE([gmpy2], [SAGE_PYTHON_PACKAGE_CHECK([gmpy2])])
diff --git a/build/pkgs/sphinx/spkg-configure.m4 b/build/pkgs/sphinx/spkg-configure.m4
new file mode 100644
index 0000000000..e149fbb2cf
--- /dev/null
+++ b/build/pkgs/sphinx/spkg-configure.m4
@@ -0,0 +1 @@
+SAGE_SPKG_CONFIGURE([sphinx], [SAGE_PYTHON_PACKAGE_CHECK([sphinx])])
diff --git a/build/pkgs/typing_extensions/spkg-configure.m4 b/build/pkgs/typing_extensions/spkg-configure.m4
new file mode 100644
index 0000000000..d475fd8257
--- /dev/null
+++ b/build/pkgs/typing_extensions/spkg-configure.m4
@@ -0,0 +1 @@
+SAGE_SPKG_CONFIGURE([typing_extensions], [SAGE_PYTHON_PACKAGE_CHECK([typing_extensions])])
-- 
diff --git a/build/pkgs/gmpy2/install-requires.txt b/build/pkgs/gmpy2/install-requires.txt
index 51d24518ef..c25feff557 100644
--- a/build/pkgs/gmpy2/install-requires.txt
+++ b/build/pkgs/gmpy2/install-requires.txt
@@ -1,3 +1 @@
-# We would like to write gmpy2 >=2.1.0b5, but pipenv does not accept prereleases in version ranges
-# https://github.com/pypa/pipenv/issues/1760
-gmpy2 ==2.1.0rc1
+gmpy2 >=2.1
diff --git a/build/pkgs/ptyprocess/install-requires.txt b/build/pkgs/ptyprocess/install-requires.txt
index de89260872..7c336f636c 100644
--- a/build/pkgs/ptyprocess/install-requires.txt
+++ b/build/pkgs/ptyprocess/install-requires.txt
@@ -1,3 +1,3 @@
-ptyprocess ==0.5.1
+ptyprocess
 # https://trac.sagemath.org/ticket/31280#comment:42 and following
 # sagelib is not compatible with ptyprocess 0.5.2, 0.6, and 0.7
